File: TvojFilm.Services/Database/Korisnici.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TvojFilm.Services.Database
{
    public class Korisnici
    {
        [Key]
        public int KorisnikId { get; set; }
        public string Ime { get; set; } = null!;
        public string Prezime { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string Telefon { get; set; } = null!;
        public DateTime DatumRodjenja { get; set; }

        public byte[]? Slika { get; set; }

        [ForeignKey(nameof(Grad))]
        public int GradId { get; set; }
        public virtual Gradovi Grad { get; set; } = null!;

        [ForeignKey(nameof(Uloga))]
        public int UlogaId { get; set; }
        public virtual Uloge Uloga { get; set; } = null!;

        public string Username { get; set; } = null!;
        public string PasswordHash { get; set; } = null!;
        public string PasswordSalt { get; set; } = null!;
    }
}

File: tvojfilmmobile/lib/model/Korisnici.dart
import 'dart:convert';

class Korisnici {
  final int? korisnikId;
  final String? ime;
  final String? prezime;
  final String? username;
  final String? email;
  final String? telefon;
  final DateTime? datumRodjenja;
  final int? gradId;
  final int? ulogaId;

  Korisnici({
    this.korisnikId,
    this.ime,
    this.prezime,
    this.username,
    this.email,
    this.telefon,
    this.datumRodjenja,
    this.gradId,
    this.ulogaId,
  });

  factory Korisnici.fromJson(Map<String, dynamic> json) {
    return Korisnici(
      korisnikId: json["korisnikId"],
      ime: json['ime'],
      prezime: json['prezime'],
      username: json['username'],
      email: json['email'],
      telefon: json['telefon'],
      datumRodjenja: DateTime.tryParse(json['datumRodjenja']),
      gradId: json['gradId'],
      ulogaId: json['ulogaId'],
    );
  }

  Map<String, dynamic> toJson() => {
        "korisnikId": korisnikId,
        "ime": ime,
        "prezime": prezime,
        "username": username,
        "email": email,
        "telefon": telefon,
        "datumRodjenja": datumRodjenja,
        "gradId": gradId,
        "ulogaId": ulogaId,
      };
}
